blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java

```diff
@@ -343,11 +343,23 @@ private static StandardUsernamePasswordCredentials findUsernamePasswordCredentia
                         Jenkins.getAuthentication(),
                         URIRequirementBuilder.fromUri(scm.getUri()).build()),
                 CredentialsMatchers.allOf(CredentialsMatchers.withId(scm.getId()),
-                        CredentialsMatchers.anyOf(CredentialsMatchers.instanceOf(StandardUsernamePasswordCredentials.class)))
+                        CredentialsMatchers.withScope(CredentialsScope.USER))
         );
     }
 
-    private static @CheckForNull StandardUsernamePasswordCredentials findUsernamePasswordCredential(String id){
+    static StandardUsernamePasswordCredentials findUsernamePasswordCredential(String uri, String credentialId){
+        return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        StandardUsernamePasswordCredentials.class,
+                        Jenkins.getInstance(),
+                        Jenkins.getAuthentication(),
+                        URIRequirementBuilder.fromUri(uri).build()),
+                CredentialsMatchers.allOf(CredentialsMatchers.withId(credentialId),
+                        CredentialsMatchers.withScope(CredentialsScope.USER))
```

Comment:
@stephenc Given I am going to be scanning in context of domain in user store, that is blueocean generated credentials are going to be created in user store in specific domain and searched accordingly, dropping USER scope is fine.